tests/test_ml.py

```diff
@@ -88,16 +91,32 @@ def test_crossvalidate(self, trait_name, cv, scoring, classifier):
         """
         training_records, genotype, phenotype, group = self.test_load_training_files(trait_name)
         clf = classifier(verb=True, random_state=RANDOM_STATE)
-        score_target = cv_scores_trex[classifier.identifier][trait_name][cv][scoring]
         score_pred = clf.crossvalidate(records=training_records, cv=cv, scoring=scoring)[:2]
-        np.testing.assert_almost_equal(actual=score_pred, desired=score_target, decimal=1)
+        if classifier.identifier in cv_scores_trex:
+            score_target = cv_scores_trex[classifier.identifier][trait_name][cv][scoring]
+            np.testing.assert_almost_equal(actual=score_pred, desired=score_target, decimal=1)
+
+    @pytest.mark.parametrize("trait_name", trait_names, ids=trait_names)
+    @pytest.mark.parametrize("classifier", classifiers, ids=classifier_ids)
+    def test_parameter_search(self, trait_name, classifier):
+        """
+        Test randomized parameter search.
+
+        :param trait_name:
+        :param classifier:
+        :return:
+        """
+        training_records, genotype, phenotype, group = self.test_load_training_files(trait_name)
+        clf = classifier(verb=True, random_state=RANDOM_STATE)
+        clf_opt = clf.parameter_search(records=training_records, n_iter=5, return_optimized=True)
+        assert type(clf_opt) == type(clf)
 
     @pytest.mark.xfail()
```

Comment:
Why is this expected to fail? We should at least have a message. Otherwise, maintenance of this code is hard.

Reply:
I left it in because I found it in TestPICASVM 😁

Reply:
Hehe